drivers/fuel_gauge/sbs_gauge/sbs_gauge.h

```diff
@@ -47,8 +47,33 @@
 
 #define SBS_GAUGE_DELAY                     1000
 
+/*
+ * Nearly all cutoff payloads are actually a singular value that must be written twice to the fuel
+ * gauge. For the case where it's a singular value that must only be written to the fuel gauge only
+ * once, retransmitting the duplicate write has no significant negative consequences.
+ *
+ * Why not devicetree: Finding the maximum length of all the battery cutoff payloads in a devicetree
+ * at compile-time would require labyrinthine amount of macro-batics.
+ *
```

Comment:
I think this is a fine trade off for this use case.

As an alternative, you could allocate an array of `cutoff_payload` for each fuel gauge based on a devictree property length and then link to the array from the driver's static `config`.   This [example](https://github.com/zephyrproject-rtos/zephyr/blob/b42331476d3d88a16a3e82c9ac83ea7864fdfc61/drivers/gpio/gpio_nct38xx_alert.c#L132) uses the `DT_INST_FOREACH_PROP_ELEM` to allocate the array.  However, this approach would likely consume a little more memory than what you've done here.

Reply:
Ack. Will keep current solution for now. Not resolving comment so other reviewers have visibility on this.